To: vim_dev@googlegroups.com
Subject: Patch 7.3.046
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.046 (after 7.3.043)
Problem:    Can't build Ruby on MS-Windows.
Solution:   Add #ifdef, don't use WIN3264 before including vim.h.
Files:	    src/if_ruby.c


*** ../vim-7.3.045/src/if_ruby.c	2010-10-27 16:49:41.000000000 +0200
--- src/if_ruby.c	2010-10-27 17:37:05.000000000 +0200
***************
*** 14,20 ****
  #include <stdio.h>
  #include <string.h>
  
! #include "auto/config.h"
  
  #ifdef _WIN32
  # if !defined(DYNAMIC_RUBY_VER) || (DYNAMIC_RUBY_VER < 18)
--- 14,22 ----
  #include <stdio.h>
  #include <string.h>
  
! #ifdef HAVE_CONFIG_H
! # include "auto/config.h"
! #endif
  
  #ifdef _WIN32
  # if !defined(DYNAMIC_RUBY_VER) || (DYNAMIC_RUBY_VER < 18)
***************
*** 50,56 ****
  #  define RUBY_EXPORT
  # endif
  
! #ifndef WIN3264
  # include <dlfcn.h>
  # define HINSTANCE void*
  # define RUBY_PROC void*
--- 52,58 ----
  #  define RUBY_EXPORT
  # endif
  
! #if !(defined(WIN32) || defined(_WIN64))
  # include <dlfcn.h>
  # define HINSTANCE void*
  # define RUBY_PROC void*
*** ../vim-7.3.045/src/version.c	2010-10-27 17:39:00.000000000 +0200
--- src/version.c	2010-10-27 17:39:59.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     46,
  /**/

-- 
BLACK KNIGHT:  Come on you pansy!
    [hah] [parry thrust]
    [ARTHUR chops the BLACK KNIGHT's right arm off]
ARTHUR:        Victory is mine!  [kneeling]
               We thank thee Lord, that in thy merc-
    [Black Knight kicks Arthur in the head while he is praying]
                                  The Quest for the Holy Grail (Monty Python)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\        download, build and distribute -- http://www.A-A-P.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
